code/API_definitions/qod-provision.yaml

```diff
@@ -0,0 +1,877 @@
+        "200":
+          description: Returns information about certain provisioning
+          headers:
+            x-correlator:
+              $ref: "#/components/headers/x-correlator"
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/ProvisioningInfo"
+              examples:
+                PROVISIONING_AVAILABLE:
+                  $ref: "#/components/examples/PROVISIONING_AVAILABLE"
+                PROVISIONING_UNAVAILABLE:
+                  $ref: "#/components/examples/PROVISIONING_UNAVAILABLE"
+        "400":
+          $ref: "#/components/responses/Generic400"
+        "401":
+          $ref: "#/components/responses/Generic401"
+        "403":
+          $ref: "#/components/responses/Generic403"
+        "404":
+          $ref: "#/components/responses/Generic404"
+        "500":
+          $ref: "#/components/responses/Generic500"
+        "503":
+          $ref: "#/components/responses/Generic503"
+      security:
+        - openId:
+            - "qod-provisioning:device-qos:read"
+
+    delete:
+      tags:
+        - QoD Provisioning
+      summary: Deletes a QoD provisioning
+      description: |
+        Release resources related to QoS provisioning.
+
+        If the notification callback is provided and the provisioning status was `AVAILABLE`, when the deletion is completed, the client will receive in addition to the response a `PROVISIONING_STATUS_CHANGED` event with
+        - `status` as `UNAVAILABLE` and
+        - `statusInfo` as `DELETE_REQUESTED`
+        There will be no notification event if the `status` was already `UNAVAILABLE`.
+      operationId: deleteProvisioning
+      parameters:
+        - $ref: "#/components/parameters/provisioningId"
+        - $ref: "#/components/parameters/x-correlator"
+      responses:
+        "204":
+          description: Provisioning deleted
+          headers:
+            x-correlator:
+              $ref: "#/components/headers/x-correlator"
+        "202":
+          description: Deletion request accepted to be processed. It applies for an async deletion process. `status` in the response will be `AVAILABLE` with `statusInfo` set to `DELETE_REQUESTED`.
+          headers:
+            x-correlator:
+              $ref: "#/components/headers/x-correlator"
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/ProvisioningInfo"
+        "400":
+          $ref: "#/components/responses/Generic400"
+        "401":
+          $ref: "#/components/responses/Generic401"
+        "403":
+          $ref: "#/components/responses/Generic403"
+        "404":
+          $ref: "#/components/responses/Generic404"
+        "500":
+          $ref: "#/components/responses/Generic500"
+        "503":
+          $ref: "#/components/responses/Generic503"
+      security:
+        - openId:
+            - "qod-provisioning:device-qos:delete"
+
+  /device-qos/retrieve:

```

Comment:
The design guidelines are a bit vague on the correct naming convention here:

> "When the POST method is used [instead of GET], the resource in the path must be a verb (e.g. `retrieve-location` and not `location`) to differentiate from an actual resource creation."

So I would expect an endpoint which would ideally be `GET /device-qos` to be named `POST /retrieve-device-qos` according to the above statement. But maybe `POST /device-qos/retrieve` also complies as `retrieve` is a verb, and I can't imagine that a developer would interpret this as creating a `retrieve`.

I'd suggest the naming convention should be clarified with Commonalities, in case both are considered to be valid options, and they can revise the guidelines accordingly.